app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    user.increment!(:posts_counter)
+  end
```

Comment:
- [ ] **[OPTIONAL]** It would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new post has been saved to the database, then make the `update_posts_counter` method private. You can also do this in the `update_likes_counter` and `update_comments_counter` methods

```ruby
 after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```